### PR TITLE
Ros2 v0.8.0 pure pursuit

### PR DESCRIPTION
**Pull request by tkimura4**

### CODE REVIEW

**Comment by wep21** [comment]
Could you replace ament_cmake_cppcheck with ament_lint_common and fix lint?
https://github.com/tkimura4/AutowareArchitectureProposal.iv/blob/c6fa8b6beee9f930ec71fd58d5ef6d90013ddd6a/control/pure_pursuit/package.xml#L24

**Comment by tkimura4** [comment]
applied lint.

File: control/pure_pursuit/include/pure_pursuit/pure_pursuit_node.hpp [hunk 1]
```diff
@@ -27,18 +27,21 @@
  * limitations under the License.
  */
 
-#include <memory>
-#include <vector>
+#ifndef PURE_PURSUIT__PURE_PURSUIT_NODE_HPP_
+#define PURE_PURSUIT__PURE_PURSUIT_NODE_HPP_
 
-#include "boost/optional.hpp"  // To be replaced by std::optional in C++17
 #include <rclcpp/rclcpp.hpp>
```
Comment (wep21) [inline control/pure_pursuit/include/pure_pursuit/pure_pursuit_node.hpp:33]: replace <> with ""

File: control/pure_pursuit/include/pure_pursuit/pure_pursuit_node.hpp [hunk 2]
```diff
@@ -27,18 +27,21 @@
...
-
-#include "tf2_ros/transform_listener.h"
-
 #include <autoware_control_msgs/msg/control_command_stamped.hpp>
```
Comment (wep21) [inline control/pure_pursuit/include/pure_pursuit/pure_pursuit_node.hpp:34]: replace <> with "" as well as 3 headers below.

File: control/pure_pursuit/include/pure_pursuit/pure_pursuit_viz.hpp [hunk 3]
```diff
@@ -27,13 +27,18 @@
  * limitations under the License.
  */
 
-#include <vector>
+#ifndef PURE_PURSUIT__PURE_PURSUIT_VIZ_HPP_
+#define PURE_PURSUIT__PURE_PURSUIT_VIZ_HPP_
+
 #include <geometry_msgs/msg/pose_stamped.hpp>
```
Comment (wep21) [inline control/pure_pursuit/include/pure_pursuit/pure_pursuit_viz.hpp:33]: replace <> with "", also 2 headers below should be fixed.

File: control/pure_pursuit/include/pure_pursuit/util/marker_helper.hpp [hunk 4]
```diff
@@ -11,12 +11,15 @@
...
 // See the License for the specific language governing permissions and
 // limitations under the License.
-#pragma once
 
-#include <string>
+#ifndef PURE_PURSUIT__UTIL__MARKER_HELPER_HPP_
+#define PURE_PURSUIT__UTIL__MARKER_HELPER_HPP_
+
 #include <rclcpp/rclcpp.hpp>
 #include <visualization_msgs/msg/marker_array.hpp>
```
Comment (wep21) [inline control/pure_pursuit/include/pure_pursuit/util/marker_helper.hpp:19]: replace <> with "" in 2 lines above.

File: control/pure_pursuit/include/pure_pursuit/util/planning_utils.hpp [hunk 5]
```diff
@@ -12,26 +12,25 @@
...
-#include <utility>
-#include <vector>
-
-#define EIGEN_MPL2_ONLY
-#include "Eigen/Core"
-#include "Eigen/Geometry"
+#ifndef PURE_PURSUIT__UTIL__PLANNING_UTILS_HPP_
+#define PURE_PURSUIT__UTIL__PLANNING_UTILS_HPP_
 
 #include <rclcpp/rclcpp.hpp>
```
Comment (wep21) [inline control/pure_pursuit/include/pure_pursuit/util/planning_utils.hpp:18]: replace <> with "", also 6 headers below should be fixed.

File: control/pure_pursuit/include/pure_pursuit/util/tf_utils.hpp [hunk 6]
```diff
@@ -12,18 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <string>
-
-#include "boost/optional.hpp"  // To be replaced by std::optional in C++17
+#ifndef PURE_PURSUIT__UTIL__TF_UTILS_HPP_
+#define PURE_PURSUIT__UTIL__TF_UTILS_HPP_
 
 #include <rclcpp/rclcpp.hpp>
```
Comment (wep21) [inline control/pure_pursuit/include/pure_pursuit/util/tf_utils.hpp:18]: replace <> with "".

File: control/pure_pursuit/src/pure_pursuit/main.cpp [hunk 7]
```diff
@@ -29,6 +29,7 @@
  */
 
 #include <rclcpp/rclcpp.hpp>
```
Comment (wep21) [inline control/pure_pursuit/src/pure_pursuit/main.cpp:31]: replace <> with "".

**Comment by tkimura4** [comment]
Fixed include brackets.

**Comment by wep21** [comment]
Could you obtain vehicle info from vehicle info utils?
https://github.com/tkimura4/AutowareArchitectureProposal.iv/blob/30539d1da6bc6ce3b9c723579bfea6143887c240/control/pure_pursuit/src/pure_pursuit/pure_pursuit_node.cpp#L69

**Comment by tkimura4** [comment]
updated the method to get vehicle info

**Review by wep21 — APPROVED** [review]
LGTM

**Pull request merged**